birdhouse/env.local.example

```diff
@@ -91,10 +102,6 @@ export GEOSERVER_ADMIN_PASSWORD=geoserverpass
 #    ./optional-components/wps-healthchecks
 #    ./optional-components/test-weaver
 #    ./optional-components/test-geoserver-secured-access
-#    ./components/scheduler
-#    ./components/monitoring
-#    ./components/weaver
-#    ./components/cowbird
```

Comment:
Oh the optional-components were before the components on purposes, to give the components higher precedence since the later item in the list can override configs of all previous items.

But now thinking about this, this means all the optional-components were already able to override all the default components.

I guess we can keep it like this but it feels a bit weird all the optional stuffs have higher override precedence than the "good" stuff.

Reply:
I'm not sure, it is as simple as that. @mishaschwartz can confirm, since he worked on it the most.
The "inspection" order follows the list, but each indicated component will then consider all their listed dependencies to load them as well before the component itself. So, components not listed or listed later could appear before the "currently inspected" component if they were marked as a dependency. 

To get a guaranteed override, one would need to set the variable value to override using single quotes and add the variable to `DELAYED_EVAL`, or define another custom component placed last in `EXTRA_CONF_DIRS` that overrides what is needed explicitly.

Reply:
> I guess we can keep it like this but it feels a bit weird all the optional stuffs have higher override precedence than the "good" stuff.

A lot of the optional components are not actually components, they're additional settings for existing components so we _want_ them to override the existing components.

> The "inspection" order follows the list, but each indicated component will then consider all their listed dependencies to load them as well before the component itself

Yeah that's a good summary. There's a detailed description of how this is enforced here: https://github.com/bird-house/birdhouse-deploy/blob/1.42.1/CHANGES.md?plain=1#L876-L962

The general recommendation is: load the optional components after the regular components since any overrides are usually desired.